03week/towersOfHanoi.js

```diff
@@ -8,37 +8,57 @@ const rl = readline.createInterface({
 });
 
 let stacks = {
-  a: [4, 3, 2, 1],
+  a: [4,3,2,1],
   b: [],
   c: []
 };
 
+//Check for valid/basic move. Should be able to move to an empty stack, and only move the top block. {a:[4,3,2],b:[1],c[]}
+//Should be able to move to an Non-empty stack. {a:[4,3,2],b:[1],c[]}
+//Should check for an illegal move. Should not be able to stack large on small. !={a:[4,3],b:[1,2],c[]}
+//Should be able to check for a win. {a:[],b:[],c[4,3,2,1]}
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+const movePiece =(startStack, endStack)=> {
+  //movePiece function should allow the removal of last item in startStack array and place into last position of endStack array.
+  // there are other methods such as (stacks[startStack.splice(-1));, however I had trouble getting this to pas test, so I stayed with pop() method
+  return stacks[endStack].push(stacks[startStack].pop());
 
 }
-
-function isLegal() {
-  // Your code here
-
+const isLegal = (startStack, endStack) => {
+  // isLegal allows player to move to an array that has only a larger last item, and to an array that is empty
+  return stacks[endStack].push() === 0 || stacks[startStack].slice(-1)[0] < stacks[endStack].slice(-1)[0];
```

Comment:
.push() is not best practice to get the length of an array, always use array.length as it will evaluate to false if the array is empty. This also does not check if startStack is not empty. Using .slice() here is unnecessary, array[array.length - 1] is easy to read and best practice. Otherwise a pretty nice line!